### ISSUE TITLE
with X11 support disabled, build fails

### ISSUE THREAD

**Issue by okias** [intel/libva-utils#153]
```
libva-utils - 2.4.0

Libva VA-API version ............. : 1.4.0
Installation prefix .............. : /usr
Extra window systems ............. : drm wayland
Enable Unit-tests .................... : no
```

```
Making all in vendor/intel/sfcsample
VDecAccelVA.cpp:45:10: fatal error: va/va_x11.h: No such file or directory
 #include <va/va_x11.h>
          ^~~~~~~~~~~~~
compilation terminated.
```

**Comment by hmaarrfk** [comment]
that VDecAccelVA.cpp file doesn't use anything from va_x11.h, so I tried commenting it out, but then the header file in the same directory seems to use things from there. Ultimately, i just removed the whole directory from the top level Makefile.am
<details> <summary> patch </summary>
```patch
diff --git a/Makefile.am b/Makefile.am
index d28175a..7f021c7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,7 +24,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = common decode encode vainfo videoprocess vendor/intel vendor/intel/sfcsample
+SUBDIRS = common decode encode vainfo videoprocess vendor/intel
 
 if USE_X11
 SUBDIRS += putsurface
diff --git a/vendor/intel/sfcsample/VDecAccelVA.cpp b/vendor/intel/sfcsample/VDecAccelVA.cpp
index ff8d06e..c7f1c13 100755
--- a/vendor/intel/sfcsample/VDecAccelVA.cpp
+++ b/vendor/intel/sfcsample/VDecAccelVA.cpp
@@ -42,7 +42,6 @@
 #include "VDecAccelVA.h"
 #include <va/va.h>
 #include <va/va_drm.h>
-#include <va/va_x11.h>
 
 #define VASUCCEEDED(err)    (err == VA_STATUS_SUCCESS)
 #define VAFAILED(err)       (err != VA_STATUS_SUCCESS)
@@ -267,7 +266,7 @@ bool mvaccel::VDecAccelVAImpl::is_slice_mode_supported(DecodeDesc& desc)
         m_vaEntrypoint,
         &vaAttrib,
         1);
- 
+
     if(vaAttrib.value & VA_DEC_SLICE_MODE_BASE || vaAttrib.value & VA_DEC_SLICE_MODE_NORMAL)
         return true;
     else
@@ -599,7 +598,7 @@ bool mvaccel::VDecAccelVAImpl::DecodePicture()
     uint32_t file_size = m_DecodeDesc.sfc_widht * m_DecodeDesc.sfc_height * 4; //ARGB format
     fwrite(gfx_surface_buf, file_size, 1, sfc_stream);
     fclose(sfc_stream);
- 
+
     //unlock surface and clear
     unlock_surface(m_sfcIDs[0]);
     Close();
@@ -684,11 +683,11 @@ int mvaccel::VDecAccelVAImpl::create_resources()
     // Prepare VAProcPipelineParameterBuffer for decode
     VAProcPipelineParameterBuffer buffer;
     memset(&buffer, 0, sizeof(buffer));
-    
+
     m_rectSrc.x = m_rectSrc.y = 0;
     m_rectSrc.width = (uint16_t)m_DecodeDesc.width;
     m_rectSrc.height = (uint16_t)m_DecodeDesc.height;
-    
+
     buffer.surface_region = &m_rectSrc;
     buffer.output_region = &m_rectSFC;
     buffer.additional_outputs = (VASurfaceID*)&(m_sfcIDs[0]);
diff --git a/vendor/intel/sfcsample/VDecAccelVA.h b/vendor/intel/sfcsample/VDecAccelVA.h
index 3ac3297..281da5d 100755
--- a/vendor/intel/sfcsample/VDecAccelVA.h
+++ b/vendor/intel/sfcsample/VDecAccelVA.h
@@ -39,7 +39,6 @@
 #include <va/va.h>
 #include "DecodeParamBuffer.h"
 
-#include <X11/Xlib.h>
 typedef Display*     HDISP;
 
 namespace mvaccel

```

</details>

**Comment by hmaarrfk** [comment]
i guess the correct fix is in https://github.com/intel/libva-utils/pull/152/files

**Comment by hmaarrfk** [comment]
This seems to have been fixed

**Comment by Duffyx** [comment]
This reappeared with 2.6.0. I build the complete libva toolchain without X11, so libva, libva-intel-media-driver and libva-utils.  Seems like libva-utils tries an X11 enabled libva library here.

libtool: link: cc -I../common -g -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing -fstack-protector -fstack-protector-strong -o hevcencode hevcencode-hevcencode.o  -L/usr/local/lib ../common/.libs/libva-display.a -ldrm -lva-drm -lva -lpthread -lm
gmake[3]: Leaving directory '/construction/xports/multimedia/libva-utils/work/libva-utils-2.6.0/encode'
Making all in vainfo
gmake[3]: Entering directory '/construction/xports/multimedia/libva-utils/work/libva-utils-2.6.0/vainfo'
cc -DHAVE_CONFIG_H -I. -I..    -I../common -I/usr/local/include  -DLIBVA_VERSION_S="\"2.6.0\"" -Wall  -fstack-protector -O2 -pipe  -DNDEBUG -fstack-protector-strong -fno-strict-aliasing  -fstack-protector -MT vainfo-vainfo.o -MD -MP -MF .deps/vainfo-vainfo.Tpo -c -o vainfo-vainfo.o `test -f 'vainfo.c' || echo './'`vainfo.c
mv -f .deps/vainfo-vainfo.Tpo .deps/vainfo-vainfo.Po
/bin/sh ../libtool  --tag=CC   --mode=link cc -I../common -I/usr/local/include  -DLIBVA_VERSION_S="\"2.6.0\"" -Wall  -fstack-protector -O2 -pipe  -DNDEBUG -fstack-protector-strong -fno-strict-aliasing  -fstack-protector  -fstack-protector-strong  -o vainfo vainfo-vainfo.o -L/usr/local/lib -lva  ../common/libva-display.la  
libtool: link: cc -I../common -I/usr/local/include -DLIBVA_VERSION_S=\"2.6.0\" -Wall -fstack-protector -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing -fstack-protector -fstack-protector-strong -o vainfo vainfo-vainfo.o  -L/usr/local/lib ../common/.libs/libva-display.a -ldrm -lva-drm -lva
gmake[3]: Leaving directory '/construction/xports/multimedia/libva-utils/work/libva-utils-2.6.0/vainfo'
Making all in videoprocess
gmake[3]: Entering directory '/construction/xports/multimedia/libva-utils/work/libva-utils-2.6.0/videoprocess'
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vavpp.o -MD -MP -MF .deps/vavpp.Tpo -c -o vavpp.o vavpp.cpp
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppscaling_csc.o -MD -MP -MF .deps/vppscaling_csc.Tpo -c -o vppscaling_csc.o vppscaling_csc.cpp
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppdenoise.o -MD -MP -MF .deps/vppdenoise.Tpo -c -o vppdenoise.o vppdenoise.cpp
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppsharpness.o -MD -MP -MF .deps/vppsharpness.Tpo -c -o vppsharpness.o vppsharpness.cpp
mv -f .deps/vppdenoise.Tpo .deps/vppdenoise.Po
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppchromasitting.o -MD -MP -MF .deps/vppchromasitting.Tpo -c -o vppchromasitting.o vppchromasitting.cpp
mv -f .deps/vppsharpness.Tpo .deps/vppsharpness.Po
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppblending.o -MD -MP -MF .deps/vppblending.Tpo -c -o vppblending.o vppblending.cpp
mv -f .deps/vppscaling_csc.Tpo .deps/vppscaling_csc.Po
c++ -DHAVE_CONFIG_H -I. -I..  -Wall -I/usr/local/include  -I../common  -fstack-protector   -O2 -pipe -DNDEBUG -fstack-protector-strong -fno-strict-aliasing   -MT vppscaling_n_out_usrptr.o -MD -MP -MF .deps/vppscaling_n_out_usrptr.Tpo -c -o vppscaling_n_out_usrptr.o vppscaling_n_out_usrptr.cpp
vppscaling_n_out_usrptr.cpp:41:10: fatal error: 'va/va_x11.h' file not found
#include <va/va_x11.h>

**Comment by gxgani** [comment]
Also facing the same issue while building on debian (10.2). Any way to circumvent this?

**Comment by th0ma7** [comment]
Having the same issue with 2.6.0 (was building all ok using 2.5.0):
```
  CXX      vppblending.o
  CXXLD    vppblending
  CXX      vppscaling_n_out_usrptr.o
vppscaling_n_out_usrptr.cpp:41:23: fatal error: va/va_x11.h: No such file or directory
 #include <va/va_x11.h>
                       ^
compilation terminated.
make[4]: *** [Makefile:533: vppscaling_n_out_usrptr.o] Error 1
make[4]: Leaving directory '/home/th0ma7/git-ffmpeg-QS-master/spksrc/cross/libva-utils/work-evansport-6.1/libva-utils-2.6.0/videoprocess'
make[3]: *** [Makefile:463: all-recursive] Error 1
make[3]: Leaving directory '/home/th0ma7/git-ffmpeg-QS-master/spksrc/cross/libva-utils/work-evansport-6.1/libva-utils-2.6.0'
make[2]: *** [Makefile:395: all] Error 2
make[2]: Leaving directory '/home/th0ma7/git-ffmpeg-QS-master/spksrc/cross/libva-utils/work-evansport-6.1/libva-utils-2.6.0'
make[1]: *** [../../mk/spksrc.compile.mk:36: compile_target] Error 2
make[1]: Leaving directory '/home/th0ma7/git-ffmpeg-QS-master/spksrc/cross/libva-utils'
make: [../../mk/spksrc.cross-cc.mk:138: arch-evansport] Error 2 (ignored)
```

**Comment by th0ma7** [comment]
After a bit of testing the culprit looks like being hash https://github.com/intel/libva-utils/commit/456c8b913b724c0926862bdfbfbe2f1d93371afc

```
#PKG_GIT_HASH = ab0806b440100b747e11288a85f3f84ff467b0f8 - ERROR
#PKG_GIT_HASH = cbbadf0301e17ac195e58265062c7e11be7fdd47 - ERROR
#PKG_GIT_HASH = 81b5929f2b71d0b657e6ef605a522edc3bcd3e80 - ERROR
#PKG_GIT_HASH = 456c8b913b724c0926862bdfbfbe2f1d93371afc - ERROR <---
#PKG_GIT_HASH = 87d77246933431f917d32f80a2eb167911175e78 - OK
#PKG_GIT_HASH = c9fb70035d2b343df077b40bdeef189e23a6beb5 - OK
```

**Comment by th0ma7** [comment]
Possible workaround https://github.com/intel/libva-utils/pull/185
And simple patch https://github.com/intel/libva-utils/pull/185/commits/0d033c074b63ab0ef0798128b94b364ce33da93a

**Comment by okias** [comment]
fixed since version [2.7.0](https://github.com/intel/libva-utils/releases/tag/2.7.0), closing.

https://github.com/intel/libva-utils/commit/bd01ba5a6b53370ee6465f393051908f9c6ddeba
